### PR TITLE
Credsdoc

### PR DESCRIPTION
**Pull request by protatremy**
Preliminary documentation explain planned work for supporting buildbot credential management

### CODE REVIEW

File: master/docs/manual/credentialmanagement.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+Definition
+----------
+
+Buildbot need to send and/or execute commands requiring credentials.
+
+The goal is to store credentials to be use by buildbot and then sent to the worker to be executed.
+
+Those informations are stored in the Buildbot database.
+
+Credentials have to be stored in a database and not in Buildbot configuration because it is
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: Please follow 'one line per sentence' requirement.

File: master/docs/manual/credentialmanagement.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
+CREDENTIAL MANAGEMENT
+=====================
+
+Definition
+----------
+
+Buildbot need to send and/or execute commands requiring credentials.
+
+The goal is to store credentials to be use by buildbot and then sent to the worker to be executed.
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: Buildbot

File: master/docs/manual/credentialmanagement.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
+CREDENTIAL MANAGEMENT
+=====================
+
+Definition
+----------
+
+Buildbot need to send and/or execute commands requiring credentials.
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: needs

File: master/docs/manual/credentialmanagement.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
+CREDENTIAL MANAGEMENT
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: Credential Management

File: master/docs/manual/credentialmanagement.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+Those informations are stored in the Buildbot database.
+
+Credentials have to be stored in a database and not in Buildbot configuration because it is
+avaliable in source configuration.
+
+Credential case: SSH Key
+------------------------
+
+When a ssh command is launched a ssh key is needed. This key is usually stored in a
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: ditto

File: master/docs/manual/credentialmanagement.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+``ssh text file`` (like id_rsa) in the $HOME directory. The ssh password entry is done with
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]:
```
:envvar:`HOME`
```

File: master/docs/manual/credentialmanagement.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+Adding new credentials
+----------------------
+
+A data API is avaliable to create, delete and modify new credentials.
+
+During creation the data API modifies the 3 informations:
+``NAME`` and ``VALUE`` (mandatory), ``PATH`` (not mandatory, as it is not used in
+the password case).
+
+A data api function helps to register any new credential in the database.
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: API

File: master/docs/manual/credentialmanagement.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+A password could be use in a command by Buildbot.
+The password is stored with a ``name``, and a ``value``. No path is inserted.
+
+Adding new credentials
+----------------------
+
+A data API is avaliable to create, delete and modify new credentials.
+
+During creation the data API modifies the 3 informations:
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: informations -> entities

File: master/docs/manual/credentialmanagement.rst [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+Credentials UI
+--------------
+
+An UI plugin helps to enter, modify and delete the credentials.
+
+Anywhere those credentials are needed, there is a drop down list of the appropriate available
+credentials, and you just select the appropriate one.
+
+When the time comes to change the password, you just change it once or delete it.
```
Comment (sa2ajj) [inline master/docs/manual/credentialmanagement.rst]: When it's time to change...

s/you //

**Comment by sa2ajj** [comment]
Looks interesting.

I'd suggest to clean up the commits though: there's no reason to have something like "fix a typo' in a new PR.

**Comment by protatremy** [comment]
Please do not review this pull request a new one was created

**Pull request closed without merging**